src/util.ts

```diff
@@ -648,9 +647,9 @@ export function isCodespaces(): boolean {
   return !!process.env["CODESPACES"];
 }
 
-export async function getAllFilePaths(dir: vscode.Uri, filename: string): Promise<string[] | undefined> {
-  const regex: RegExp = new RegExp(`(\/|\\\\)${escapeStringRegexp(filename)}$`);
-  return recGetAllFilePaths(dir.fsPath, filename, regex, fs.readdirSync(dir.fsPath), []);
+export async function getAllCMakeListsPaths(dir: vscode.Uri): Promise<string[] | undefined> {
+  const regex: RegExp = new RegExp(/(\/|\\)CMakeLists\.txt$/);
+  return recGetAllFilePaths(dir.fsPath, "CMakeLists.txt", regex, fs.readdirSync(dir.fsPath), []);
```

Comment:
I don't think you should need to pass in "CMakeLists.txt" as the second parameter.  It doesn't look like it's used.  Can you just remove the 2nd parameter from the recursive function?

Reply:
And should the 4th parameter be `await readDir(dir.fsPath)`?